proxy/http/parser/parser.py

```diff
@@ -301,16 +301,18 @@ def _process_line_and_headers(self, raw: bytes) -> Tuple[bool, bytes]:
 
     def _process_line(self, raw: bytes) -> None:
         if self.type == httpParserTypes.REQUEST_PARSER:
+            # parse request line according https://datatracker.ietf.org/doc/html/rfc2616#section-5.1
             if self.protocol is not None and self.protocol.version is None:
                 # We expect to receive entire proxy protocol v1 line
                 # in one network read and don't expect partial packets
                 self.protocol.parse(raw)
             else:
                 line = raw.split(WHITESPACE)
-                self.method = line[0].upper()
-                self.set_url(line[1])
-                self.version = line[2]
-                self.state = httpParserStates.LINE_RCVD
+                if len(line) == 3:
```

Comment:
Thinking more about it.  I feel may be we must just raise an exception when line doesn't have sufficient parts as per the specification.  Reason why I think raising an exception is better than silently ignoring invalid packets:

1. Issue #127 is generally not an issue for a well behaved clients
2. This is however a possible attack vector because malicious users/hackers can start sending specially crafted request to the server.
3. If we choose to ignore invalid bytes, a hacker can still overwhelm the server by streaming invalid chunks to the server.
4. E.g. imagine, someone by mistake connects a SSH client on port `8899`.  What will happen then?  Will our parser keep on rejecting invalid bytes, expecting to see a valid HTTP packet someday?   May be answer is to drop such invalid clients as soon as possible.

I think we can simply raise a `NotImplementedError` for now which will eventually result in dropping the client connection from the core.   Wdyt?

Reply:
That's right, we should quick fail other than waiting for valid input